docs/sql-guide/statements/statement-insert-bulk.md

```diff
@@ -0,0 +1,165 @@
+| `'Inline_stream'` | The contents of the literal read as though they were in a file.  | [FROM inline stream examples](/docs/sql-guide/statements/statement-insert-bulk/#from-inline-stream) |
+| `WITH` | Pass one or more statement level options. |  |
+| `BATCHSIZE` | Specify the batch size of the BULK commit. Defaults to 1000. |  |
+| `ROWSLIMIT` | Limit the number of rows processed in a batch. |
+| `INPUT` | Set the type of input to `'FILE'`, `'URL'` or `'STREAM'`. |  |
+| `FORMAT` | Set the format of the source data to `'csv'` or `'ndjson'`. |  |
+| `HEADER_ROW` | Optional CSV argument that will ignore the header in the source CSV file. |  |
+| `ALLOW_MISSING_VALUES` | Optional argument that overrides a `NULL` error message that will stop processing when valid JsonPath expressions in the MAP clause have missing data. |  |
+
+## Errors
+
+| Error | Override |
+|---|---|
+| Data in source column cannot be converted to specified data type | None |
+| Mismatch between expressions in the column list and TRANSFORM clause | None |
+| NULL error when valid JsonPath expression in MAP clause have missing data | ALLOW_MISSING_VALUES |
+
+## TRANSFORM clause
+
+* a list of valid SQL expressions that are used to specify data transformation before values are inserted.
+* uses variables named for the ordinal position values are specified in the MAP clause.
+
+{: .important}
+The number of expressions in the column list and TRANSFORM clause must match.
+
+* [TRANSFORM example](/docs/sql-guide/statements/statement-insert-bulk/#transform-examples)
+
+## Examples
+
+### MAP examples
+
+| Input type | MAP expression for value in source column | Example |
+|---|---|---|
+| CSV | Integer offset |  |
+| NDJSON | String [JsonPath expression](https://goessner.net/articles/JsonPath/index.html#e2) for the NDSON value | `MAP ('$.idpath' id, '$.int.path.' int, '$.path.to.string' string)` |
+
+### TRANSFORM examples
+
+| Map clause | TRANSFORM clause |
+|---|---|
+| `MAP (0 id, 1 int, 4 string)` | Variables: `@0`, `@1` and `@2` |
+
+```sql
+TRANSFORM (
+    @0 + 10, -- offset the new _id value by 10
+    @1,      -- pass through unchanged
+    case     -- clean up state names
+        when @2 = 'Texas' then 'TX' end
+        when @2 = 'California' then 'CA' end
+        else @2
+    end
+)
+```
+
+### FROM quotation examples
+
+FROM clause quotation marks must be escaped before the BULK statement is run, even when CSV values are quoted.
+
+| Incorrect | Correct |
+|---|---|
+| `FeatureBase's speed` | `FeatureBase''s speed` |
+| `""Time is money." – Benjamin Franklin."` | `"""Time is money."" – Benjamin Franklin."` |
+
+### FROM inline stream
+
+The contents of an inline stream string literal are treated as a file and read line-by-line.
+
+Single line:
+
+```
+'this is a single-line string literal'
+```
+
+Multi line (prepend with `x`)
+
+```sql
+x'this
+is
+a
+multi-line
+string
+literal'
+```
+
+### BULK REPLACE from CSV file with TRANSFORM
+
+```sql
+bulk replace
```

Comment:
this doesn't exist yet so let's not have this unless you got it from somewhere?

Reply:
I confess I have no idea where this came from. Commented out but can just trash if easier :D